pkg/networkservice/common/crossnse/server.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package crossnse provides a NetworkServiceServer chain element that tracks local Cross connect Endpoints and call them first
+// their unix file socket as the clienturl.ClientURL(ctx) used to connect to them.
+package crossnse
+
+import (
+	"context"
+	"net/url"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/trace"
+	"github.com/networkservicemesh/sdk/pkg/registry/common/crossnse"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/clienturl"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+type crossConnectNSEServer struct {
+	// Map of names -> *registry.NetworkServiceEndpoint for local bypass to file crossNSEs
+	crossNSEs sync.Map
```

Comment:
Was there a particular reason not to generate typed sync.Maps similar to https://github.com/networkservicemesh/sdk/blob/master/pkg/tools/clientmap/gen.go ?